Source/SwiftLintFramework/Rules/Idiomatic/ExplicitTypeInterfaceRule.swift

```diff
@@ -130,6 +132,23 @@ private extension Dictionary where Key == String, Value == SourceKitRepresentabl
         return initCallRegex.firstMatch(in: contentAfterName, options: [], range: contentAfterName.fullNSRange) != nil
     }
 
+    func isTypeReferenceAssignment(file: File) -> Bool {
+        guard
+            let nameOffset = nameOffset,
+            let nameLength = nameLength,
+            case let contents = file.contents.bridge(),
+            let afterNameRange = contents.byteRangeToNSRange(start: nameOffset + nameLength, length: 0)
+        else {
+            return false
+        }
+
+        let contentAfterName = contents.substring(from: afterNameRange.location)
+        let typeAssignment =
+            regex("^\\s*=\\s*(?:try[!?]?\\s+)?\\[?\\p{Lu}[^\\(\\s<]*(?:<[^\\>]*>)?(?::\\s*[^\\(\\n]+)?\\]?self")
```

Comment:
Can this be simplified? 😲
For example, we don't need to worry about `try` in this case.

Also, shouldn't it enforce a `.` before `self`?

Reply:
I simplified it and made sure there's a `.` before the `self`.